config/environments/development.rb

```diff
@@ -1,47 +1,84 @@
+  # Show full error reports.
+  config.consider_all_requests_local = true
 
-  # Log error messages when you accidentally call methods on nil.
-  # config.whiny_nils = true
+  # Enable/disable caching. By default caching is disabled.
+  # Run rails dev:cache to toggle caching.
+  if Rails.root.join("tmp/caching-dev.txt").exist?
+    config.action_controller.perform_caching = true
+    config.action_controller.enable_fragment_cache_logging = true
 
-  # Show full error reports:
-  config.consider_all_requests_local = true
+    memcached_servers = "127.0.0.1:11211"
+    memcached_servers = YAML.load_file(Rails.root.join("config/local.yml")).fetch("MEMCACHED_SERVERS", memcached_servers) if File.file?(Rails.root.join("config/local.yml"))
+    config.cache_store = :mem_cache_store, memcached_servers,
+                         { namespace: "ao3-v2-dev", compress: true, pool_size: 10 }
+  else
+    config.action_controller.perform_caching = false
+
+    config.cache_store = :null_store
+  end
 
-  # Enable/disable caching. By default caching is disabled, but it can be
-  # toggled on and off by calling rails dev:cache and restarting the server.
-  config.action_controller.perform_caching = Rails.root.join("tmp/caching-dev.txt").exist?
+  # Store uploaded files on the local file system (see config/storage.yml for options).
+  config.active_storage.service = :local
 
-  # Don't care if the mailer can't send
+  # Don't care if the mailer can't send.
   config.action_mailer.raise_delivery_errors = false
 
+  config.action_mailer.perform_caching = false
+
   # Enable mailer previews at http://localhost:3000/rails/mailers.
   config.action_mailer.show_previews = true
 
-  # Print deprecation notices to the Rails logger
+  # Print deprecation notices to the Rails logger.
   config.active_support.deprecation = :log
 
-  # Only use best-standards-support built into browsers
-  config.action_dispatch.best_standards_support = :builtin
+  # Raise exceptions for disallowed deprecations.
+  config.active_support.disallowed_deprecation = :raise
+
+  # Tell Active Support which deprecation messages to disallow.
+  config.active_support.disallowed_deprecation_warnings = []
+
+  # Raise an error on page load if there are pending migrations.
+  config.active_record.migration_error = :page_load
+
+  # Highlight code that triggered database queries in logs.
+  config.active_record.verbose_query_logs = true
+
+  # Debug mode disables concatenation and preprocessing of assets.
+  # This option may cause significant delays in view rendering with a large
+  # number of complex assets.
+  config.assets.debug = true
+
+  # Suppress logger output for asset requests.
+  config.assets.quiet = true
+
+  # Raises error for missing translations.
+  config.i18n.raise_on_missing_translations = true
+
+  # Annotate rendered view with file names.
+  # config.action_view.annotate_rendered_view_with_filenames = true
```

Comment:
!!!

Oh, my god, this seems like it's going to be so useful. (But I agree it's a good idea to have it off by default.)